### DISCUSSION TITLE
VitePress equivalent of extendPageData from VuePress

### DISCUSSION THREAD

**Discussion by erondpowell** [vuejs/vitepress#1066] [Q&A]
Is there is a VitePress equivalent for Vuepress's `extendPageData` function?

In my project, we need data to generate page thumbnail previews (via [Thumbsmith](https://thumbsmith.com/)) for each docs page.
To do this, I need to assign and use data for each page *at build time*.

In VuePress, there was an `extendPageData` function:
https://vuepress.vuejs.org/plugin/option-api.html#extendpagedata

I see VitePress has a `useData` function:
https://vitepress.vuejs.org/guide/api.html#usedata
It appears that this is executed at runtime, client-side.
If that's correct, I don't think this would work in our use-case.

What are my best options?

**Comment by brc-dd** [comment]
> Is there is a VitePress equivalent for Vuepress's extendPageData function?

No.

> What are my best options?

I'm assuming you need to get some data (say from front matter), and add head tags for `og:image`. You can use our `transformHtml` hook (it'll be there in the next release). But it'll not change that image when you move page to page (client side). Search engines and social media platforms will pick them up correctly though because the og:image tag will be there in the sent html. It's still complex and low level.

Another alternative will be available when #551 is resolved. There you'll get an API to modify your head content on build.

**Reply by erondpowell** [reply]
Yeah basically. We wanted to access and add values to the VitePress equivalent of the `$page` object:
https://vuepress.vuejs.org/guide/global-computed.html#site
Then use that data to generate a custom image thumbnail (via thumbsmith) for each page.
Sounds like it cannot be done at this point.
Thanks for the quick reply!
